### PR TITLE
Rewrite part5

### PR DESCRIPTION
**Pull request by shortdudey123**
Rebase of https://github.com/sous-chefs/mongodb/pull/8 on the `rewrite_v1` branch

### CODE REVIEW

File: libraries/mongodb.rb [hunk 1]
```diff
@@ -27,13 +27,8 @@ def self.configure_replicaset(node, name, members)
...
     require 'mongo'
 
-    if members.empty?
-      if Chef::Config[:solo]
-        Chef::Log.warn('Cannot search for member nodes with chef-solo, defaulting to single node replica set')
-      else
-        Chef::Log.warn("Cannot configure replicaset '#{name}', no member nodes found")
-        return
-      end
+    if members.empty? && Chef::Config[:solo]
```
Comment (damacus) [inline libraries/mongodb.rb]: Solo checks shouldn't be here. Solo/zero now supports search.

Comment (rshade) [inline libraries/mongodb.rb]: check out the haproxy cookbook, we created a test with it.

Comment (damacus) [inline libraries/mongodb.rb]: @rshade the haprpxy cookbook has a solo check in it? We should fix zis

Comment (shortdudey123) [inline libraries/mongodb.rb]: looks the the haproxy cookbook has searches using zero not solo unless i am missing something

Comment (rshade) [inline libraries/mongodb.rb]: @damacus it looks like it will error if we don't have this.

**Pull request merged**